src/dataProvider/SparqlDataProvider.js

```diff
@@ -365,3 +379,50 @@ async function configureBindingStream(bindingStream, variables) {
     throw new HttpError(error.message, 500);
   }
 }
+
+const addComunicaContextSourcesFromSourcesIndex = async (sourcesIndex) => {
+  const sourcesList = [];
+  try {
+    const result = await fetch(`${config.queryFolder}${sourcesIndex.queryLocation}`);
+    const queryStringIndexSource = await result.text();
+
+    const bindingsStream = await myEngine.queryBindings(queryStringIndexSource, {
+      sources: [sourcesIndex.url],
+    });
+
+    await new Promise((resolve, reject) => {
+      bindingsStream.on('data', (binding) => {
+        const source = binding.get('object').value;
+        if (!sourcesList.includes(source)) {
+          sourcesList.push(source);
+        }
+      });
+      bindingsStream.on('end', resolve);
+      bindingsStream.on('error', reject);
+    });
+  } catch (error) {
+    console.error("Error adding sources from index:", error);
```

Comment:
Remove the console.error call. Compose a new Error from your text here and the underlying error message text.